addons/account_peppol/models/peppol_constraints.py

```diff
@@ -0,0 +1,34 @@
+import re
+from stdnum import get_cc_module, ean
+
+
+PEPPOL_ENDPOINT_RULES = {
+    '0007': ['se', 'orgnr'],
+    '0088': ['ean'],
+    '0184': ['dk', 'cvr'],
+    '0192': ['no', 'orgnr'],
+    '0208': ['be', 'vat'],
+}
+
+PEPPOL_ENDPOINT_WARNING = {
+    '0201': ['regex', '[0-9a-zA-Z]{6}$'],
+    '0210': ['it', 'codicefiscale'],
+    '0211': ['it', 'iva'],
+    '9906': ['it', 'iva'],
+    '9907': ['it', 'codicefiscale'],
+    '0151': ['au', 'abn'],
+}
+
+def _check_peppol_endpoint_number(peppol_eas, peppol_endpoint):
+    endpoint_rule = PEPPOL_ENDPOINT_RULES.get(peppol_eas) or PEPPOL_ENDPOINT_WARNING.get(peppol_eas)
+    if not endpoint_rule:
+        return False
+
+    if endpoint_rule[0] == 'regex':
+        return bool(re.match(endpoint_rule[1], peppol_endpoint))
+
+    if endpoint_rule[0] == 'ean':
+        check_module = ean
+    else:
+        check_module = get_cc_module(endpoint_rule[0], endpoint_rule[1])
+    return check_module.is_valid(peppol_endpoint)
```

Comment:
This should be in account_edi_ubl_cii because we will need that for partners as well

Reply:
we don't want to add a constraint on partners, because we don't have control over how partners have been registered on the network by some other provider. If it's wrong, but they are registered, there is nothing we could do about that. And we have the required verification, so if the user entered the endpoint incorrectly, the partner won't be verified